### PR TITLE
Add monitoring labs

### PR DESCRIPTION
**Pull request by mabegglen**

### CODE REVIEW

File: content/en/docs/03/02_monitoring-configuration.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+             storage: 5Gi
+...
+```
+
+It is advisable to persist the Alertmanager also. This allows Alertmanager to retain active alerts on restarts and therefore prevents triggering existing alerts.
+
+```bash
+oc -n openshift-monitoring edit cm cluster-monitoring-config
+```
+
```
Comment (bliemli) [inline content/en/docs/03/02_monitoring-configuration.md:54]:
```suggestion

Add the following config:

```

File: content/en/docs/03/02_monitoring-configuration.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+```yaml
+...
```
Comment (bliemli) [inline content/en/docs/03/02_monitoring-configuration.md:56]:
```suggestion
```

File: content/en/docs/03/_index.md [hunk 3]
```diff
@@ -4,3 +4,8 @@ weight: 30
 sectionnumber: 3
 ---
 
+OpenShift provides a cluster monitoring stack, which can be used for [monitor default platform components](https://docs.openshift.com/container-platform/4.7/monitoring/understanding-the-monitoring-stack.html) or [user-defined projects components](https://docs.openshift.com/container-platform/4.7/monitoring/enabling-monitoring-for-user-defined-projects.html).
+
+The OpenShit provided [cluster logging](https://docs.openshift.com/container-platform/4.7/logging/cluster-logging.html) uses ElasticSearch to store the cluster logs, which then made be available to users using Kibana.
+
+In this lab you are going to configure alerting for the cluster monitoring stack, enabling monitoring for user-defined projects and deploy the logging stack.
```
Comment (bliemli) [inline content/en/docs/03/_index.md:11]:
```suggestion
OpenShift provides a cluster monitoring stack which can be used to [monitor default platform components](https://docs.openshift.com/container-platform/latest/monitoring/understanding-the-monitoring-stack.html) or even [user-defined projects components](https://docs.openshift.com/container-platform/latest/monitoring/enabling-monitoring-for-user-defined-projects.html).

The OpenShit-provided [cluster logging](https://docs.openshift.com/container-platform/latest/logging/cluster-logging.html) uses ElasticSearch to store cluster and application logs, which can then be visualised using Kibana.

In this lab, you are going to configure alerting for the cluster monitoring stack, enable monitoring for user-defined projects and deploy the logging stack.
```

File: content/en/docs/03/03_user-defined-rules.md [hunk 4]
```diff
@@ -4,3 +4,80 @@ weight: 33
 sectionnumber: 3.3
 ---
 
+## Task {{% param sectionnumber %}}.1: Enable the user-workload monitoring stack
+
+As mentioned in a lab before, it is not supported to alter the OpenShift provided `openshift-monitoring` stack. If you want to add additional monitoring targets or add custom Prometheus rules, you need to enable the user-workload monitoring stack.
```
Comment (bliemli) [inline content/en/docs/03/03_user-defined-rules.md:9]:
```suggestion
As mentioned in a lab before, it is not supported to alter the OpenShift-provided monitoring stack. If you want to add additional monitoring targets or add custom Prometheus rules, you need to enable the user-workload monitoring stack.

And this is exactly what we are going to do. First, edit the `cluster-monitoring-config` configmap:
```

File: content/en/docs/03/03_user-defined-rules.md [hunk 5]
```diff
@@ -4,3 +4,80 @@ weight: 33
...
+
+```bash
+oc -n openshift-monitoring edit cm cluster-monitoring-config
+```
+
```
Comment (bliemli) [inline content/en/docs/03/03_user-defined-rules.md:14]:
```suggestion

Add the line `enableUserWorkload: true` under `data/config.yaml` (leave the rest as it is):

```

File: content/en/docs/03/03_user-defined-rules.md [hunk 6]
```diff
@@ -4,3 +4,80 @@ weight: 33
...
+```yaml
+...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+...
```
Comment (bliemli) [inline content/en/docs/03/03_user-defined-rules.md:24]:
```suggestion
```

File: content/en/docs/03/02_monitoring-configuration.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+## Task {{% param sectionnumber %}}.1: Enable alerting
+
+Let's take a look at an example of how you can send alerts to a receiver of your choice. You can do so by editing the settings directly in the web console at **Administration** -> **Cluster Settings** -> **Global Configuration** -> **Alertmanager** or by creating the alertmanager-main secret.
+
+```bash
+# Alertmanager example with dummy SMTP server
+cat <<EOF | oc -n openshift-monitoring replace -f -
+{{< readfile file="content/en/docs/03/resources/alertmanager-main.yaml" >}}
+EOF
+```
```
[the remark this answers is not in the document]
Comment (mabegglen) [reply content/en/docs/03/02_monitoring-configuration.md]: fixed in eff6a3cfeefdcc544ca97944164425b3e1c7ca68

File: content/en/docs/03/02_monitoring-configuration.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+Lets take a look at the main components that can be configured applying a custom Alertmanager configuration.
+
+* `receivers`: With a receiver, one or more notification types can be defined. e.g. mail, webhook, or one of the message platforms like Slack or PagerDuty.
+* `routes`: With routing blocks, a tree of routes and child routes can be defined. Each routing block has a matcher which can match none or several labels of an alert. Per block, one receiver can be specified, or if empty, the default receiver is taken.
+  * `Watchdog`: There is a default Watchdog alert that is constantely firing to make sure, that the complete alerting pipeline is functional. It is best-practice to implement a Deadman's switch on the receiving side to get notified, when sending alerts should fail.
+* `inhibit_rules`: You can group alerts together to suppress certain alerts to prevent notification flooding. E.g. when an etcd member is unavailable (**severity: critical**) you do not care if the etcd has a high number of failed leader proposals (**severity: warning**).
```
[the remark this answers is not in the document]
Comment (mabegglen) [reply content/en/docs/03/02_monitoring-configuration.md]: fixed in eff6a3cfeefdcc544ca97944164425b3e1c7ca68

File: content/en/docs/03/03_user-defined-rules.md [hunk 9]
```diff
@@ -4,3 +4,80 @@ weight: 33
...
+
+## Task {{% param sectionnumber %}}.2: Add a custom Prometheus rule
+
+With a configured user-workload monitoring stack you can add your own custom Prometheus rules. To do so, create a PrometheusRule custom resource.
+
+```bash
+cat <<EOF | oc -n uptime-app-prod apply -f -
+{{< readfile file="content/en/docs/03/resources/custom-prometheus-rule.yaml" >}}
+EOF
+```
```
[the remark this answers is not in the document]
Comment (mabegglen) [reply content/en/docs/03/03_user-defined-rules.md]: fixed in eff6a3cfeefdcc544ca97944164425b3e1c7ca68

**Pull request merged**